pitmutationmate/src/main/kotlin/com/amos/pitmutationmate/pitmutationmate/listener/BuildFileListener.kt

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023 Lennart Heimbs
+
+package com.amos.pitmutationmate.pitmutationmate.listener
+
+import com.amos.pitmutationmate.pitmutationmate.buildSystem.BuildSystemUtils
+import com.amos.pitmutationmate.pitmutationmate.services.PluginCheckerService
+import com.intellij.openapi.components.service
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.fileEditor.FileDocumentManagerListener
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.ProjectManager
+import com.intellij.openapi.vfs.VirtualFile
+
+class BuildFileListener : FileDocumentManagerListener {
+    private val logger = Logger.getInstance(this::class.java)
+
+    override fun beforeDocumentSaving(document: Document) {
+        FileDocumentManager.getInstance().getFile(document)?.let { file ->
+            ProjectManager.getInstance().openProjects.forEach { project ->
+                if (isVirtualFileInProject(file, project)) {
+                    logger.debug("Project: ${project.name} is being checked for plugins because it contains ${file.path}")
+                    project.service<PluginCheckerService>().checkPlugins(file)
```

Comment:
```suggestion
                    val pluginChecker = project.service<PluginCheckerService>()
                    pluginChecker.reset()
                    pluginChecker.checkPlugins(file)
```

Reply:
Sorry for the confusion this should be part of the review below.

Reply:
Why? The whole point is to have state in that object so that the files dont need to be rechecked all the time

Reply:
Ah ok then I misundertstood it. I thought it should check the files whenever something changed.

Reply:
It's checking a specific file through the file listener when it changes. And upon that the respective kotlin or gradle availability object is overwritten.